### PR TITLE
5.x: upgrade to PHPUnit 10

### PR DESCRIPTION
**Pull request by LordSimal**
Refs: https://github.com/cakephp/cakephp/pull/16992/

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -41,8 +41,8 @@
         "psr-4": {
             "Cake\\Test\\": "vendor/cakephp/cakephp/tests/",
             "DebugKit\\Test\\": "tests/",
-            "DebugKit\\TestApp\\": "tests/test_app/",
-            "DebugkitTestPlugin\\": "tests/test_app/Plugin/DebugkitTestPlugin/src/"
+            "DebugKit\\TestApp\\": "tests/TestApp/",
```
Comment (ADmad) [inline composer.json]: I prefer having `test_app` over `TestApp`.

File: src/ToolbarService.php [hunk 2]
```diff
@@ -102,7 +102,7 @@ public function registry(): PanelRegistry
      */
     public function isEnabled(): bool
     {
-        if (isset($GLOBALS['__PHPUNIT_BOOTSTRAP'])) {
+        if (defined('PHPUNIT_COMPOSER_INSTALL') && !$GLOBALS['FORCE_DEBUGKIT_TOOLBAR']) {
```
Comment (markstory) [inline src/ToolbarService.php:105]: Of course they break my hacks :laughing:

Comment (LordSimal) [reply src/ToolbarService.php:105]: Just re-introduced a new hack 😁

File: tests/TestCase/Middleware/DebugKitMiddlewareTest.php [hunk 3]
```diff
@@ -125,7 +125,7 @@ public function testInvokeSaveData()
         $this->assertNotNull($result->panels[11]->summary);
         $this->assertSame('Sql Log', $result->panels[11]->title);
 
-        $timeStamp = filemtime(Plugin::path('DebugKit') . 'webroot' . DS . 'js' . DS . 'main.js');
+        $timeStamp = filectime(Plugin::path('DebugKit') . 'webroot' . DS . 'js' . DS . 'main.js');
```
Comment (garas) [inline tests/TestCase/Middleware/DebugKitMiddlewareTest.php:128]: It should be `inject-iframe.js` here and keep using `filemtime`.

Line was previously incorrectly updated when .js file was renamed. And `ToolbarService::getToolbarUrl‎` is using `filemtime`.

Comment (LordSimal) [reply tests/TestCase/Middleware/DebugKitMiddlewareTest.php:128]: I tried that but for some reason its not working... haven't looked too deep into it though.

**Pull request merged**